spec/frise/loader_spec.rb

```diff
@@ -75,28 +62,153 @@ def validators.short_string(_, str)
       str.is_a?(String) && str.length < 5 || (raise "expected a short string, found #{str.inspect}")
     end
 
-    loader = Loader.new(
-      schema_load_paths: [fixture_path('_schemas')],
-      validators: validators
-    )
+    loader = Loader.new(validators: validators, exit_on_fail: true)
 
-    expect { loader.load(fixture_path('loader_test_1.yml'), true) }.to output(
+    expect { loader.load(fixture_path('loader_test1_schema.yml')) }.to output(
       "1 config error(s) found:\n" \
       " - At int: missing required value\n"
     ).to_stdout.and raise_error(SystemExit)
   end
 
   it 'should use an extra symbol table when provided' do
-    loader = Loader.new(
-      schema_load_paths: [fixture_path('_schemas')],
-      defaults_load_paths: [fixture_path('_defaults')]
+    loader = Loader.new(exit_on_fail: false)
+
+    conf = loader.load(fixture_path('loader_test2_all.yml'), '_id' => 'myobj')
+    expect(conf).to eq(
+      'id' => 'myobj',
+      'name' => 'My Object',
+      'description' => 'Description of My Object'
+    )
+  end
+
+  it 'should allow using a different key or no key for inclusions and schemas' do
+    loader = Loader.new(include_sym: '__custom_inc', schema_sym: '__custom_sch', exit_on_fail: false)
+
+    conf = loader.load(fixture_path('loader_test2_all_alt.yml'), '_id' => 'myobj')
+    expect(conf).to eq(
+      'id' => 'myobj',
+      'name' => 'My Object',
+      'description' => 'Description of My Object'
+    )
+
+    loader = Loader.new(include_sym: nil, schema_sym: nil, exit_on_fail: false)
+
+    conf = loader.load(fixture_path('loader_test1_all.yml'))
+    expect(conf).to eq(
+      '$schema' => [fixture_path('_schemas/loader_test1.yml')],
+      '$include' => [fixture_path('_defaults/loader_test1.yml')],
+      'str' => 'str',
+      'bool' => true,
+      'arr' => %w[elem0 elem1],
+      'obj' => { 'key2' => 'value2' },
+      'record' => { 'field1' => 'f1', 'field2' => true }
+    )
+  end
+
+  it 'should process inclusions and schemas recursively' do
+    loader = Loader.new(exit_on_fail: false)
+
+    conf = loader.load(fixture_path('loader_test3.yml'))
+    expect(conf).to eq(
+      'id' => 'myobj',
+      'name' => 'My Object',
+      'description' => 'Description of My Object',
+      'age' => 20
+    )
+  end
+
+  it 'should process inclusions and schemas deep in the config, providing a _this variable in defaults' do
+    loader = Loader.new(exit_on_fail: false)
+
+    conf = loader.load(fixture_path('loader_test4.yml'))
+    expect(conf).to eq(
+      'key1' => {
+        'value1_1' => 1,
+        'value1_2' => 'abc',
+        'key2' => { 'value2_1' => 2 }
+      }
+    )
+  end
+
+  it 'should process multiple inclusions and schemas in the same position' do
```

Comment:
What should happen when the next include redefines some schema or default?

Reply:
The includes are merged in order, so the original file has priority over the first include and the first include has priority over the second. That's tested in this spec.